src/sympc/tensor/share_tensor.py

```diff
@@ -91,7 +91,9 @@ def sanity_checks(
         """
         if op_str == "mul" and isinstance(y, (float, torch.FloatTensor)):
             y = ShareTensor(data=y, session=x.session)
-        elif op_str in {"add", "sub", "lt", "gt"} and not isinstance(y, ShareTensor):
+        elif op_str in {"add", "sub", "lt", "gt", "matmul"} and not isinstance(
```

Comment:
I'm not sure of this, especially because I'm unsure of what `if op_str == "mul" and isinstance(y, (float, torch.FloatTensor)):` really does

Reply:
Here I was thinking about the following scenario: if we multiply and is a ```LongTensor``` then we should not convert it, but I think this adds more complexity (as you pointer out here :D)

```
if not isinstance(y, ShareTensor):
   y = ShareTensor
```

What do you think, @LaRiffle ?